chat.go

```diff
@@ -51,7 +56,31 @@ type PromptAnnotation struct {
 	ContentFilterResults ContentFilterResults `json:"content_filter_results,omitempty"`
 }
 
+type ChatMessageImageURL struct {
+	URL string `json:"url"`
+}
+
+type ChatMessageContent struct {
+	Type     string               `json:"type"`
+	Text     string               `json:"text,omitempty"`
+	ImageURL *ChatMessageImageURL `json:"image_url,omitempty"`
+}
+
 type ChatCompletionMessage struct {
+	Role    string               `json:"role"`
+	Content []ChatMessageContent `json:"content"`
```

Comment:
This decision is not within my jurisdiction, but these modifications breaks compatibility. I have already suggested a less intrusive alternative [here](https://github.com/sashabaranov/go-openai/issues/539#issuecomment-1799826250). I am awaiting the maintainer's involvement to facilitate a discussion on the matter.

Reply:
The OpenAI API still accepts string values for Content. Ideally this wrapper conforms to their API contract.

Reply:
@sashabaranov 
Do you have any idea how to deal with backward compatibility? I summed-up the possible solutions (that I see) here.